tool/tsh/resolve_default_addr_test.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+var testLog = log.WithField(trace.Component, "test")
+
+func newWaitForeverHandler() (http.Handler, chan interface{}) {
+	doneChannel := make(chan interface{})
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		testLog.Debug("Waiting forever...")
+		<-doneChannel
+	})
+
+	return handler, doneChannel
+}
+
+func newRespondingHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		testLog.Debug("Responding")
+		w.Header().Add("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusOK)
+		io.WriteString(w, "Hello, world")
+	})
+}
+
+func mustGetCandidatePorts(servers []*httptest.Server) []int {
+	result := make([]int, len(servers))
+	for i, svr := range servers {
+		u, err := url.Parse(svr.URL)
+		if err != nil {
+			panic(err)
+		}
+
+		p, err := strconv.Atoi(u.Port())
+		if err != nil {
+			panic(err)
+		}
+
+		result[i] = p
+	}
+	return result
+}
+
+func Test_ResolveDefaultAddr(t *testing.T) {
```

Comment:
Nit: We don't usually separate Test with an underscore from the test name. Not huge deal, but I would try to be consistent in naming.

Reply:
Done.